prince-of-versions/src/main/java/co/infinum/princeofversions/PrinceOfVersionsConfig.java

```diff
@@ -167,7 +167,11 @@ public Builder withOptionalNotificationType(NotificationType optionalNotificatio
          * @return this builder
          */
         public Builder withMetadata(Map<String, Object> metadata) {
-            this.metadata = metadata;
+            if (this.metadata == null) {
+                this.metadata = metadata;
```

Comment:
Be careful here. Just setting variable and later on adding all to the same map will show the changes in that map also if user uses that later in the code.
My suggestion is to construct empty map when creating Builder and to always putAll in it.